core/src/Streamly/Internal/Data/MutArray/Generic.hs

```diff
@@ -778,7 +784,7 @@ chunksOf n (D.Stream step state) =
                 putIndexUnsafeWith end contents x
                 let end1 = end + 1
                 return $
-                    if end1 >= bound
+                    if start + end1 >= bound
```

Comment:
BUG: this should be `end1 >= bound`. end1 is the end index and bound is the max index + 1.

Reply:
Looks like the previous code was bugged.